crmsh/bootstrap.py

```diff
@@ -1742,16 +1749,27 @@ def update_expected_votes():
         # Increase expected_votes
         # TODO: wait to adjust expected_votes until after cluster join,
         # so that we can ask the cluster for the current membership list
+        # Have to check if a qnetd device is configured and increase
+        # expected_votes in that case
+        qnetd_votes = 1 if corosync.get_value("quorum.device.model") == "net" else 0
```

Comment:
Hi, Kristoffer 
Sorry to make qdevice unclear. 

in fact the qnetd_votes is the same thing with device_votes, 

without qdevice, normally expected_vote is number of nodes.
assume we have a cluster with N nodes. expected_votes will be N without qdevice.
But qdevice, expected_vote is sum of "number of nodes" and device_votes.

there are two algorithms so far, ffsplit and lms

for ffsplit,  device_votes is always on, so the expected_votes is N + 1.
as described on bsc#1070961 comment#15, we will not add the device_votes to expected_votes at first init,
but  on the first join(indeed the second node join the cluster) by judging "device_votes==0 and expected_votes==1", we can set  device_votes to 1, while adding device_votes to expected_votes only this time.

Reply:
while for lms, device_votes is N - 1, and the expected_votes is (N-1) + N
we can get number of nodes: num = (expected_votes+1)/2,  
and increase number of nodes: num += 1
get the new  device_votes:  device_votes = num - 1
and get the new expected_votes:  expected_votes = (num -1) + num

this is recalculated every time a node joins